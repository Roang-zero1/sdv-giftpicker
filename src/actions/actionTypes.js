export const INCREMENT_ITEM_COUNT = 'INCREMENT_ITEM_COUNT';
export const SET_LOADED = 'SET_LOADED';
export const UPDATE_ITEMS = 'UPDATE_ITEMS';
export const SET_PROGRESS = 'SET_PROGRESS';
export const SET_GIFT_COUNT = 'SET_GIFT_COUNT';
