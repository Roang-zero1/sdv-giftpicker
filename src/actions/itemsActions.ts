import { createStandardAction } from 'typesafe-actions';
import { IItems, IModifyItemCount } from '../common/types';

export const modifyItemCount = createStandardAction('items/MODIFY_COUNT')<
  IModifyItemCount
>();

export const updateItems = createStandardAction('items/UPDATE_ITEMS')<IItems>();
