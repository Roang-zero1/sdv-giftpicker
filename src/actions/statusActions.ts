import { createStandardAction } from 'typesafe-actions';

export const setSaveGame = createStandardAction('status/SET_SAVE_GAME')<
  boolean
>();

export const setIntroChosen = createStandardAction('status/SET_INTRO_CHOSEN')<
  boolean
>();

export const setLoading = createStandardAction('status/SET_LOADING')<boolean>();
