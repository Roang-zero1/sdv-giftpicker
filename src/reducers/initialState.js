export default {
  characters: {},
  status: { intro: false, save: false }
};
