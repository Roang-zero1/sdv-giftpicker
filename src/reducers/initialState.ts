import { initialState as initialCharactersState } from './charactersReducer';
import { initialState as initialItemsState } from './itemsReducer';
import { initialState as initialNavigationState } from './navigationReducer';
import { initialState as initialStatusState } from './statusReducer';

export default {
  characters: initialCharactersState,
  items: initialItemsState,
  navigation: initialNavigationState,
  status: initialStatusState
};
