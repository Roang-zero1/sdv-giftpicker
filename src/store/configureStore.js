import { persistReducer, persistStore } from 'redux-persist';

import { createStore } from 'redux';
import rootReducer from '../reducers/rootReducer';
import storage from 'redux-persist/lib/storage';

const persistConfig = {
  key: 'sdv-gp',
  storage,
  blacklist: ['navigation']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
export const persistor = persistStore(store);
