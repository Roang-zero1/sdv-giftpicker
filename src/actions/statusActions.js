import * as types from './actionTypes';

export function setSaveGame(save = false) {
  return { type: types.SET_SAVE_GAME, save: save };
}

export function setIntroChosen(intro = false) {
  return { type: types.SET_INTRO_CHOSEN, intro: intro };
}

export function setLoading(loading = false) {
  return {
    type: types.SET_LOADING,
    loading: loading
  };
}

export function setFluidLayout(fluid = false) {
  return { type: types.SET_FLUID_LAYOUT, fluid };
}
