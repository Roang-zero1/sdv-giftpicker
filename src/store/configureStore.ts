import { PersistConfig, persistReducer, persistStore } from 'redux-persist';

import { createStore, StoreEnhancer } from 'redux';
import storage from 'redux-persist/lib/storage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers/rootReducer';

const persistConfig: PersistConfig = {
  blacklist: ['navigation'],
  key: 'sdv-gp',
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const devTools: StoreEnhancer = (window as any).__REDUX_DEVTOOLS_EXTENSION__
  ? (window as any).__REDUX_DEVTOOLS_EXTENSION__()
  : (f: any) => f;

// @ts-ignore Currently bugged for redux 4.0.0
export const store = createStore(persistedReducer, initialState, devTools);
export const persistor = persistStore(store);
