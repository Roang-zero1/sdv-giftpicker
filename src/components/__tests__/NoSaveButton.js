import ConnectedNoSaveButton, { NoSaveButton } from '../NoSaveButton';
import { mount, shallow } from 'enzyme';

import { Provider } from 'react-redux';
import React from 'react';
import configureStore from 'redux-mock-store';
import giftIDs from '../../data/Gifts';
import initialState from '../../reducers/initialState';
import toJSON from 'enzyme-to-json';
import * as statusActions from '../../actions/statusActions';
import * as itemsActions from '../../actions/itemsActions';
import { isActionOf } from 'typesafe-actions';

describe('components/NoSaveButton --- Shallow render component', () => {
  let cut;

  beforeEach(() => {
    cut = shallow(<NoSaveButton />);
  });

  it('should render the one component', () => {
    expect(cut).toHaveLength(1);
  });

  it('should be the same as the last snapshot', () => {
    expect(toJSON(cut)).toMatchSnapshot();
  });
});

describe('components/NoSaveButton --- Render component', () => {
  let renderedCut;
  let props;
  const cut = () => {
    if (!renderedCut) {
      renderedCut = mount(<NoSaveButton {...props} />);
    }
    return renderedCut;
  };

  beforeEach(() => {
    renderedCut = undefined;
    props = {
      inline: undefined,
      text: undefined
    };
  });

  it('should render the one component', () => {
    expect(cut()).toHaveLength(1);
  });

  describe('when `text` is defined ', () => {
    beforeEach(() => {
      props.text = 'Remove Save Game';
      cut();
    });

    it('should render the passed text', () => {
      expect(cut().prop('text')).toBe(props.text);
      expect(
        cut()
          .find('Button')
          .text()
      ).toEqual(props.text);
    });
  });

  describe('when `inline` is defined ', () => {
    beforeEach(() => {
      props.inline = true;
    });

    it('should add a css class when inline', () => {
      expect(
        cut()
          .find('button')
          .hasClass('btn-sm')
      ).toEqual(true);
    });
  });
});

describe('components/NoSaveButton --- Shallow render connected component', () => {
  const mockStore = configureStore();
  let cut;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    cut = mount(
      <Provider store={store}>
        <ConnectedNoSaveButton />
      </Provider>
    );
  });

  it('should render the one component', () => {
    expect(cut).toHaveLength(1);
  });

  it('should handle the onClick action correctly', () => {
    expect(store.getState()).toEqual(initialState);
    cut.find('button').simulate('click');
    let actions = store.getActions();
    let setLoading = actions.filter(isActionOf(statusActions.setLoading));
    expect(setLoading).toHaveLength(2);
    expect(setLoading[0].payload).toEqual(true);
    expect(setLoading[1].payload).toEqual(false);

    let setSaveGame = actions.filter(isActionOf(statusActions.setSaveGame));
    expect(setSaveGame).toHaveLength(1);
    expect(setSaveGame[0].payload).toEqual(false);

    let setIntroChosen = actions.filter(
      isActionOf(statusActions.setIntroChosen)
    );
    expect(setIntroChosen).toHaveLength(1);
    expect(setIntroChosen[0].payload).toEqual(true);

    expect(
      Object.keys(
        actions.filter(isActionOf(itemsActions.updateItems))[0].payload
      )
    ).toHaveLength(giftIDs.length);
  });
});
