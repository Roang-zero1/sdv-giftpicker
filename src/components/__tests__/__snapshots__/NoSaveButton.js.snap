// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NoSaveButton --- Snapshot should be the same as the last snapshot 1`] = `
<button
  className="btn btn-secondary"
  onClick={[Function]}
  type="button"
>
  Use without save
</button>
`;
